src/db.rs

```diff
@@ -96,6 +96,7 @@ impl std::hash::Hasher for IdentityKeyHash {
 struct DbInner {
 	columns: Vec<Column>,
 	options: Options,
+	metadata: Metadata,
```

Comment:
ColumnOptions being stored in both metadata and options, does not look problematic.

Reply:
Semantically columns settings in options have a meaning of what the database metadata should be when it is created anew. Might be a good idea to split `Db::open` into `create` that takes column settings and `open` that does not.